pkg/crc/pullsecret/pullsecret.go

```diff
@@ -0,0 +1,176 @@
+package pullsecret
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/code-ready/crc/pkg/crc/errors"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/systemd"
+	"github.com/code-ready/machine/libmachine/drivers"
+	"github.com/pborman/uuid"
+)
+
+var (
+	pullSecret = `apiVersion: v1
+data:
+  .dockerconfigjson: %s
+kind: Secret
+metadata:
+  name: pull-secret
+  namespace: openshift-config
+type: kubernetes.io/dockerconfigjson`
+
+	// This command timeout in 80 secs if not able to replace the pull secret.
+	replacePullSecretCmd = `timeout 80 bash -c 'until oc --config /tmp/kubeconfig replace -f /tmp/pull-secret.yaml 2>/dev/null 1>&2; \
+do echo "Waiting for recovery apiserver to come up."; sleep 1; done'`
+
+	// This command is used to update the clusterID using random uuid.
+	// https://stackoverflow.com/a/1250279
+	updateClusterIdCmd = `timeout 80 bash -c 'until oc --config /tmp/kubeconfig patch clusterversion version -p '"'"'{"spec":{"clusterID":"%s"}}'"'"' --type merge 2>/dev/null 1>&2; \
+do echo "Waiting for recovery apiserver to come up."; sleep 1; done'`
+
+	// This command make sure we stop the kubelet and clean up the pods
+	// We also providing a 2 sec sleep so that stop pods get settled and
+	// ready for remove. Without this 2 sec time sometime it happens some of
+	// the pods are not completely stopped and when remove happens it will throw
+	// an error like below.
+	// remove /var/run/containers/storage/overlay-containers/97e5858e610afc9f71d145b1a7bd5ad930e537ccae79969ae256636f7fb7e77c/userdata/shm: device or resource busy
+	stopAndRemovePodsCmd = `bash -c 'sudo crictl stopp $(sudo crictl pods -q) && sleep 2 &&\
+sudo crictl rmp $(sudo crictl pods -q)'`
+)
+
+func AddPullSecretAndClusterID(driver drivers.Driver, pullSec string, kubeconfigFilePath string) error {
+	// Add the kubeconfig File to the Instance.
+	if err := addKubeconfigFileToInstance(driver, kubeconfigFilePath); err != nil {
+		return err
+	}
+
+	// Add the Pull secret kubernetes resource definition to Instance
+	if err := addpullSecretSpecToInstance(driver, pullSec); err != nil {
+		return err
+	}
+
+	// Replace user pull secret and add cluster ID
+	if err := setPullSecretAndClusterID(driver); err != nil {
+		return err
+	}
+
+	// Add user pull secret to the instance
+	if err := addpullSecretToIntanceDisk(driver, pullSec); err != nil {
```

Comment:
Still "Intance" typo.

Reply:
Any reason for tying `setPullSecretAndClusterID` together?
I would write this as:
```
if err := addKubeconfigFileToInstance(driver, kubeconfigFilePath); err != nil {
        return err
}

if err := setPullSecret(driver, pullSec); err != nil {
//...
}

if err := setClusterID(driver); err != nil {
//...
}
```

Reply:
> Any reason for tying setPullSecretAndClusterID together?

@cfergeau yes so I don't have to start the kubelet service 2 times and the clean up again :(

Reply:
I had something like this in mind:
https://github.com/cfergeau/crc/commit/6355e185bd9c27b87e705c733ecc5d7d4e210a38 (only compile tested)

Reply:
@cfergeau Initially I had same codebase then when I tried to run it and check the how many pods run in a given span of time is way too much.  When I was using a single script which was doing `kubelet start` => changing all the bits => `kubelet stop` => removing the pods that time only 4 pods which got created during that timeframe now after we made it moduler now around 8 pods are running in that time frame and if we again putting the defer on top of `AddPullSecretAndClusterID` then we might hit 10-12 pods and each time we delay we are adding more delay in clean up step so I didn't consider this idea.

Reply:
hmm ok, I did not realize this before now. The whole point of making the command modular was to be able to group related things together. If we can not move things around freely because otherwise the cleanup becomes too costly, then we could probably have kept the 2 bash commands together.